CRM/Member/BAO/Membership.php

```diff
@@ -2125,9 +2125,12 @@ public static function setQuickConfigMembershipParameters($membershipOrg, $membe
    */
   protected static function updateDeceasedMembersStatuses() {
     $count = 0;
+
+    $deceasedStatusId = CRM_Core_PseudoConstant::getKey('CRM_Member_BAO_Membership', 'status_id', 'Deceased');
+
     // 'create' context for buildOptions returns only if enabled.
     $allStatus = self::buildOptions('status_id', 'create');
-    if (($deceasedStatusId = array_search('Deceased', $allStatus)) === FALSE) {
+    if (array_key_exists($deceasedStatusId, $allStatus) === FALSE) {
```

Comment:
I think just if ($deceasedStatusId) would work wouldn't it - it returns false if no key is found

Reply:
In my case, `CRM_Core_PseudoConstant::getKey()` returns a value for a membership’s `status_id` even if not enabled. That’s why we have to use the `buildOptions()` with the `create` argument.

Here is an example with the “New” (“Nouveau” in French) status which is disabled in my configuration:

```
$ cv cli
>>> CRM_Core_PseudoConstant::getKey('CRM_Member_BAO_Membership', 'status_id', 'New');
=> 1
>>> CRM_Member_BAO_Membership::buildOptions('status_id', 'create');
=> [
     2 => "Courant",
     3 => "Délai de grâce",
     4 => "Expiré",
     5 => "En instance",
     6 => "Annulé",
     7 => "Décédé",
   ]
>>>
```